src/components/Dashboard/ThisWeek/NewsletterMenu/NewsletterMenu.test.tsx

```diff
@@ -0,0 +1,177 @@
+        GetTaskAnalyticsQuery & GetEmailNewsletterContactsQuery
+      >>
+        <NewsletterMenu accountListId={accountListId} />
+      </GqlMockedProvider>,
+    );
+
+    expect(queryByText('Never')).toBeNull();
+    expect(queryByText('Newsletter')).toBeVisible();
+  });
+
+  it('should open newsletter menu', async () => {
+    const { queryByText, queryByTestId } = render(
+      <GqlMockedProvider<
+        GetTaskAnalyticsQuery & GetEmailNewsletterContactsQuery
+      >>
+        <NewsletterMenu accountListId={accountListId} />
+      </GqlMockedProvider>,
+    );
+
+    userEvent.click(queryByTestId('NewsletterMenuButton'));
+
+    expect(queryByText('Log Newsletter')).toBeInTheDocument();
+    expect(queryByText('Export Email')).toBeInTheDocument();
+    expect(queryByText('Export Physical')).toBeInTheDocument();
+  });
+
+  describe('Log Newsletter', () => {
+    it('should open Log Newsletter menu', () => {
+      const { queryByText, queryByTestId } = render(
+        <GqlMockedProvider<
+          GetTaskAnalyticsQuery & GetEmailNewsletterContactsQuery
+        >>
+          <NewsletterMenu accountListId={accountListId} />
+        </GqlMockedProvider>,
+      );
+
+      userEvent.click(queryByTestId('NewsletterMenuButton'));
+      userEvent.click(queryByText('Log Newsletter'));
+
+      expect(
+        queryByText('Log Newsletter placeholder text'),
+      ).toBeInTheDocument();
+    });
+  });
+
+  describe('Export Email', () => {
+    it('should open Export Email dialog', async () => {
+      const email1 = 'fakeemail1@fake.com';
+      const email2 = 'fakeemail2@fake.com';
+
+      const mocks = {
+        GetEmailNewsletterContacts: {
+          contacts: {
+            nodes: [
+              {
+                primaryPerson: {
+                  primaryEmailAddress: {
+                    email: email1,
+                  },
+                },
+              },
+              {
+                primaryPerson: {
+                  primaryEmailAddress: {
+                    email: email2,
+                  },
+                },
+              },
+            ],
+          },
+        },
+      };
+      const { queryByText, queryByTestId } = render(
+        <GqlMockedProvider<
+          GetTaskAnalyticsQuery & GetEmailNewsletterContactsQuery
+        >
+          mocks={mocks}
+        >
+          <NewsletterMenu accountListId={accountListId} />
+        </GqlMockedProvider>,
+      );
+
+      userEvent.click(queryByTestId('NewsletterMenuButton'));
+      userEvent.click(queryByText('Export Email'));
+
+      expect(queryByText('Copy All')).toBeInTheDocument();
+      await waitFor(() => expect(queryByTestId('emailList')).not.toBeNull());
+      expect(queryByTestId('emailList')).toHaveValue(`${email1},${email2}`);
+    });
+  });
+
+  it('should copy email list to clipboard', async () => {
+    Object.assign(navigator, {
+      clipboard: {
+        writeText: () => {},
+      },
+    });
```

Comment:
I had to do some googling to figure out how to test copying text to the clipboard. This is what I based this on: https://stackoverflow.com/questions/62351935/how-to-mock-navigator-clipboard-writetext-in-jest